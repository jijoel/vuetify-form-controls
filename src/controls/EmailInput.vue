<template>
<v-text-field
  :value="value"
  v-bind="$attrs"
  v-on="$listeners"
  @input="input"
  @blur="blur"
  type="email"
  :name="name"
  :label="label"
  :browser-autocomplete="browserAutocomplete"
  :prepend-icon="prependIcon"
></v-text-field>
</template>

<script>
import ChildControl from '../mixins/ChildControl'

export default {

  name: 'EmailInput',

  mixins: [ ChildControl ],

  props: {
    name: {
      required: false,
      default: 'email'
    },
    browserAutocomplete: {
      required: false,
      default: 'email'
    },
    label: {
      required: false,
      default: 'E-Mail Address'
    },
    prependIcon: {
      required: false,
      default: 'email'
    },
  },

}
</script>
