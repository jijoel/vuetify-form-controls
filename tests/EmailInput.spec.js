import './setup'
import { common } from './common'
import { mount, shallowMount } from '@vue/test-utils'
import EmailInput from '../src/controls/EmailInput'

describe('EmailInput', () => {

  common(EmailInput)

})
