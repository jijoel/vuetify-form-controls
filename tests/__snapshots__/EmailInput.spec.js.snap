// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput is rendered as custom vuetify text field 1`] = `
<v-text-field-stub
  browser-autocomplete="email"
  label="E-Mail Address"
  name="email"
  prepend-icon="email"
  type="email"
  value=""
/>
`;

exports[`EmailInput renders properties sent by parent 1`] = `
<v-text-field-stub
  :solo="true"
  browser-autocomplete="email"
  label="E-Mail Address"
  name="email"
  prepend-icon="email"
  type="email"
  value=""
/>
`;
